### DISCUSSION TITLE
how to disable events from downloading artifacts and reconciliation

### DISCUSSION THREAD

**Discussion by mo-saeed** [fluxcd/flux2#2896] [General]
how to disable events from downloading artifacts and reconciliation ?

**Comment by kingdonb** [comment]
We do not have a way to silence events at present, unless you're talking about notifications from the notification controller – in that case you can filter them out at the Alert level. You can change the event alert sources to ignore the noisy source, or you can filter by regex.

Which controller events are the noisy ones that you want to remove? Depending on what kind of event you want to silence we can recommend something.

**Comment by mo-saeed** [comment]
Hello @kingdonb,

Thank you for your answers, I am talking about kubernetes events 

```
4m3s        Normal    ArtifactUpToDate             helmrepository/oauth2-proxy                                      artifact up-to-date with remote revision: 'af1bc798f1b18087d6b4dfc816c7b2ba51bd86759063cb370aee5b812a1a52ff'
54m         Normal    ReconciliationSucceeded      kustomization/cluster-kustomization                              Reconciliation finished in 1m29.159460125s, next run in 10m0s
```

**Reply by kingdonb** [reply]
There is no mechanism to squelch Kubernetes events, that I am aware of. Can you elaborate further on why it is that you want these silenced? They tell about what the resource is doing, and they are meant to be informative about how the resource spends its time and what lifecycle events there are. If the events are happening too frequently, you could turn up the interval to something longer and then instrument these HelmRepository and other sources with a Notification Receiver (webhook)

Is it an issue of something that is consuming the events/logs and the cost is too high, like stackdriver? It would be good to know more about what motivates this need to be able to answer better about what remediations are possible. Flux emits events, and some events may be required for certain API integrations to be able to work in the future. So I do not foresee the Flux APIs adding a configurable threshold for which events get emitted, but if these notifications are noisy and unhelpful maybe they do not make it into the 2.0.0 GA release. (In principle any breaking changes could land sooner or later but once 2.0.0 is landed, it becomes harder to make such a change. So this type of feedback is really welcome and especially valuable now, opposed to later than when a GA Flux v2 is released.)

**Comment by mo-saeed** [comment]
@kingdonb exactly it's cost issue, aall the events are forwarded to azure log workspace, so we need to pay for the data transfer and this is a lot especially that we are applying reconciliation in very short interval. 

I mean we can increase the interval abit but will not help a lot. we would like to see when reconciliation fails so only show warnings and errors.

**Reply by kingdonb** [reply]
Thank you for clarifying. I think the best I can suggest for now is filtering, if there was any opportunity to simply not log those events, or filter those events from the log, but it sounds like that wouldn't stop them from being delivered necessarily so your concerns about data transfer costs are sustained.

Please open an issue on the notification controller with details about the chatty alert so we can discuss the best way to remediate this issue, if it affects your production then I think it's reasonable to ask for a mode where "all OK, no changes" is not echoed into the logs every interval of activity. Perhaps it should be removed, or maybe there can be a less chatty mode added. Either way as it sounds important to you it's likely important to others, so please open an issue there.
